### PR TITLE
Create requered models by g model

### PR DESCRIPTION
**Pull request by elmar8287**
- Create model classes for all entities as shown in the ERD diagram.
- Set up associations between models.
- The models should include some additional methods.

- Users:

A method that returns the 3 most recent posts for a given user.

- Posts:

A method that updates the posts counter for a user.
A method that returns the 5 most recent comments for a given post.

- Comments:

A method that updates the comments counter for a post.

- Likes:

A method that updates the likes counter for a post.

### CODE REVIEW

File: app/models/comment.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class Comment < ApplicationRecord
+  belongs_to :user
+  belongs_to :post
+
+  def update_comments_counter
+    post.increment!(:commentsCounter)
+  end
```
Comment (DeliceLydia) [inline app/models/comment.rb:7]: ## Optional
- [x] I suggest that you use one of the Rails Active Callbacks - `after_save` to call the `#update_comments_counter` method every time a post receive a comment, then make the `update_comments_counter` method private

File: app/models/like.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class Like < ApplicationRecord
+  belongs_to :user
+  belongs_to :post
+
+  def update_likes_counter
+    post.increment!(:likesCounter)
+  end
```
Comment (DeliceLydia) [inline app/models/like.rb:7]: ## Optional
- [x] I suggest that you use one of the Rails Active Callbacks - `after_save` to call the `#update_likes_counter`  every time a post receive a like, then make the `update_likes_counter` method private

File: app/models/post.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :user
+  has_many :comments
+  has_many :likes
+
+  def update_posts_counter
+    user.increment!(:postsCounter)
+  end
```
Comment (DeliceLydia) [inline app/models/post.rb:8]: ## Optional
- [x] I suggest that you use one of the Rails Active Callbacks - `after_save` to call the `#update_posts_counter` method every time a new post has been saved to the database, then make the `update_posts_counter` method private

**Review by DeliceLydia — APPROVED** [review]
# Congratulations🎉🎉

## Hi @elmar8287, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard).. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by elmar8287** [comment]
Thanks for the review! 👍

**Pull request merged**